### PR TITLE
[SPARK-28814][SQL][DOC] Document SET/RESET in SQL Reference

### PR DESCRIPTION
**Pull request by sharangk**
### What changes were proposed in this pull request?
Document SET/REST statement in SQL Reference Guide.

### Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

### Does this PR introduce any user-facing change?
Yes.

#### Before:
There was no documentation for this.

#### After:

**SET**
image
image

**RESET**
image

### How was this patch tested?
Manual Review and Tested using jykyll build --serve

### CODE REVIEW

**Comment by sharangk** [comment]
@dilipbiswal , @gatorsmile , Please review. Thanks

**Comment by sharangk** [comment]
@dilipbiswal , @huaxingao , @gatorsmile , Could you please review and let me know your comments.

File: docs/sql-ref-syntax-aux-conf-mgmt-set.md [hunk 1]
```diff
@@ -19,4 +19,51 @@ license: |
   limitations under the License.
 ---
 
-**This page is under construction**
+### Description
+The SET command sets a property, returns the value of an existing property or returns all SQLConf properties with value and meaning.
+
+### Syntax
+{% highlight sql %}
+SET;
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-aux-conf-mgmt-set.md]: Can we remove the semi colon from the syntax ?

File: docs/sql-ref-syntax-aux-conf-mgmt-set.md [hunk 2]
```diff
@@ -19,4 +19,51 @@ license: |
...
+SET [ -v ];
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-aux-conf-mgmt-set.md]: ditto

File: docs/sql-ref-syntax-aux-conf-mgmt-set.md [hunk 3]
```diff
@@ -19,4 +19,51 @@ license: |
...
+SET property_key[ =property_value ];
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-aux-conf-mgmt-set.md]: ditto.  Add a space after equal ?

File: docs/sql-ref-syntax-aux-conf-mgmt-set.md [hunk 4]
```diff
@@ -19,4 +19,51 @@ license: |
...
+</dl>
+
+<dl>
+  <dt><code><em>property_key</em></code></dt>
+  <dd>Returns the value of specified property key.</dd>
+</dl>
+
+<dl>
+  <dt><code><em>property_key=property_value</em></code></dt>
+  <dd>Sets the value of specified property key.</dd>
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-aux-conf-mgmt-set.md]: Can we add that , "The new value overrides any existing value for the specified property key" ?

**Review by dilipbiswal — CHANGES_REQUESTED** [review]

File: docs/sql-ref-syntax-aux-conf-mgmt-reset.md [hunk 5]
```diff
@@ -19,4 +19,20 @@ license: |
   limitations under the License.
 ---
 
-**This page is under construction**
+### Description
+Reset all SQLConf properties to the default values. After RESET command, executing SET command will output empty.
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-aux-conf-mgmt-reset.md]: Can we say "Reset all the properties specific to the current session to their default values" ? I would like to remove SQLConf as even non sql properties are being cleared.

**Comment by sharangk** [comment]
@dilipbiswal , Thanks for the review. Resubmitted the PR with updates. Please have a look. Thanks !!!

File: docs/sql-ref-syntax-aux-conf-mgmt-set.md [hunk 6]
```diff
@@ -42,7 +42,7 @@ SET property_key[ =property_value ];
 
 <dl>
   <dt><code><em>property_key=property_value</em></code></dt>
-  <dd>Sets the value of specified property key.</dd>
+  <dd>The new value overrides any existing value for the specified property key.</dd>
```
Comment (dilipbiswal) [inline docs/sql-ref-syntax-aux-conf-mgmt-set.md]: @sharangk This makes it sound like it only supports a update op ? How about the following ?
```
Sets the value for a given property key. If an old value exists for a given property key, then it gets overridden by the new value.
```

**Comment by dilipbiswal** [comment]
LGTM - except a minor comment.

File: docs/sql-ref-syntax-aux-conf-mgmt-set.md (hunk 6 above)
Comment (sharangk) [reply docs/sql-ref-syntax-aux-conf-mgmt-set.md]: Done. Thanks.

**Comment by sharangk** [comment]
@dilipbiswal , Thanks. Handled the review comments.

**Comment by dilipbiswal** [comment]
LGTM
cc @gatorsmile

**Comment by gatorsmile** [comment]
ok to test

**Review by gatorsmile — COMMENTED** [review]
LGTM

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**